exporter/signalfxexporter/factory.go

```diff
@@ -22,8 +22,10 @@ import (
 )
 
 const (
-	defaultHTTPTimeout = time.Second * 10
-	defaultMaxConns    = 100
+	defaultHTTPTimeout          = time.Second * 10
+	defaultHTTP2ReadIdleTimeout = time.Second * 10
+	defaultHTTP2PingTimeout     = time.Second * 5
```

Comment:
Would 10s be a safer default (still 5s less than the transport default)? In the dead connection case we potentially add 5s to data loss window but in normal operation we're using half the other default timeout. I could see this leading to connection churn when the service is already degraded.

Reply:
> Would 10s be a safer default (still 5s less than the transport default)?
It's actually 5 seconds more than the transport default, as you have to add the (ping + read idle) = 15 seconds

> I could see this leading to connection churn when the service is already degraded.
Are you concerned that the system might be under heavy load, causing I/O slow down/15 seconds halt and resulting of unnecessary connection closure? 

In case you missed my note in the description:
```
As to why I chose 10 seconds for HTTP2ReadIdleTimeout and 5 seconds for HTTP2PingTimeout
Those values have been tested in production and they will result, in an active env (with default http timeout of 10 seconds), of a single export failure at max before the health check detects the corrupted tcp connection and closes it.
The only drawback is if the connection was not used for over 10 seconds, we might end up sending unnecessary ping frames, which should not be an issue and if it became an issue, then we can tune those settings.
```

Reply:
After chatting with Ryan, it made sense to be on the safe side and avoid any unseen issue with the short timeout .
Setting the ping timeout to 10 s as suggested